rafx-api/src/backends/gles2/internal/gl_context/native.rs

```diff
@@ -0,0 +1,1175 @@
+            self.gles2.BindVertexArray(self.global_vao);
+            self.check_for_error().unwrap();
+        }
+    }
+
+    pub fn make_not_current(&self) {
+        unsafe {
+            self.gles2.BindVertexArray(0);
+            self.check_for_error().unwrap();
+        }
+        self.context.make_not_current();
+    }
+
+    pub fn swap_buffers(&self) {
+        self.context.swap_buffers();
+    }
+
+    pub fn check_for_error(&self) -> RafxResult<()> {
+        check_for_error(&self.gles2)
+    }
+
+    pub fn gl_get_integerv(
+        &self,
+        pname: u32,
+    ) -> i32 {
+        unsafe {
+            let mut value = 0;
+            self.gles2.GetIntegerv(pname, &mut value);
+            value
+        }
+    }
+
+    pub fn gl_get_string(
+        &self,
+        pname: u32,
+    ) -> String {
+        unsafe {
+            let str = self.gles2.GetString(pname);
+            if str.is_null() {
+                return "".to_string();
+            }
+
+            std::ffi::CStr::from_ptr(str as _)
+                .to_str()
+                .unwrap()
+                .to_string()
+        }
+    }
+
+    pub fn gl_viewport(
+        &self,
+        x: i32,
+        y: i32,
+        width: i32,
+        height: i32,
+    ) -> RafxResult<()> {
+        unsafe {
+            self.gles2.Viewport(x, y, width, height);
+            self.check_for_error()
+        }
+    }
+
+    pub fn gl_scissor(
+        &self,
+        x: i32,
+        y: i32,
+        width: i32,
+        height: i32,
+    ) -> RafxResult<()> {
+        unsafe {
+            self.gles2.Scissor(x, y, width, height);
+            self.check_for_error()
+        }
+    }
+
+    pub fn gl_depth_rangef(
+        &self,
+        n: f32,
+        f: f32,
+    ) -> RafxResult<()> {
+        unsafe {
+            self.gles2.DepthRangef(n, f);
+            self.check_for_error()
+        }
+    }
+
+    pub fn gl_clear_color(
+        &self,
+        r: f32,
+        g: f32,
+        b: f32,
+        a: f32,
+    ) -> RafxResult<()> {
+        unsafe {
+            self.gles2.ClearColor(r, g, b, a);
+            self.check_for_error()
+        }
+    }
+
+    pub fn gl_clear_depthf(
+        &self,
+        d: f32,
+    ) -> RafxResult<()> {
+        unsafe {
+            self.gles2.ClearDepthf(d);
+            self.check_for_error()
+        }
+    }
+
+    pub fn gl_clear_stencil(
+        &self,
+        s: i32,
+    ) -> RafxResult<()> {
+        unsafe {
+            self.gles2.ClearStencil(s);
+            self.check_for_error()
+        }
+    }
+
+    pub fn gl_clear(
+        &self,
+        mask: u32,
+    ) -> RafxResult<()> {
+        unsafe {
+            self.gles2.Clear(mask);
+            self.check_for_error()
+        }
+    }
+
+    pub fn gl_finish(&self) -> RafxResult<()> {
+        unsafe {
+            self.gles2.Finish();
+            self.check_for_error()
+        }
+    }
+
+    pub fn gl_create_framebuffer(&self) -> RafxResult<FramebufferId> {
```

Comment:
In this file, do we care about naming consistency with the GL functions?
`gl_gen_framebuffer` would be closer.

Reply:
Ditto for buffers /textures + the destroy vs delete.

Reply:
The APIs for these functions do not identically match the corresponding GL functions. Additionally, the function names/args do not match between web-sys and gl.